lib/spl/src/spl_manager.c

```diff
@@ -0,0 +1,429 @@
+#include "spl_manager.h"
+
+#include <nitro/fx/fx.h>
+#include <nnsys/gfd/VramManager/gfd_PlttVramMan_Types.h>
+#include <nnsys/gfd/VramManager/gfd_TexVramMan_Types.h>
+
+#include "spl_behavior.h"
+#include "spl_emitter.h"
+#include "spl_internal.h"
+#include "spl_particle.h"
+
+#define DECODE_WH(X) ((u16)(1 << ((X) + 3)))
+#define EMITTER_SHOULD_TERMINATE(emtr, header)                                                                                         \
+    ((                                                                                                                                 \
+         (header->flags.selfMaintaining && header->emitterLifeTime != 0 && emtr->state.started && emtr->age > header->emitterLifeTime) \
+         || emtr->state.terminate)                                                                                                     \
+        && emtr->particles.count == 0 && emtr->childParticles.count == 0)
```

Comment:
**suggestion:** Either format the parens in multi-line style or pull the second line here up to the first line.